[CI] ShardChangesIT#testFollowIndex_backlog

```
FAILURE 11.2s J1 | ShardChangesIT.testFollowIndex_backlog <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <2258L>
   >      but: was <2300L>
   >    at __randomizedtesting.SeedInfo.seed([DC85FF589F9FC48A:65A7ED6E8C945A2]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.xpack.ccr.ShardChangesIT.lambda$assertTotalNumberOfOptimizedIndexing$31(ShardChangesIT.java:929)
   >    at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:847)
   >    at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:821)
   >    at org.elasticsearch.xpack.ccr.ShardChangesIT.assertTotalNumberOfOptimizedIndexing(ShardChangesIT.java:913)
   >    at org.elasticsearch.xpack.ccr.ShardChangesIT.testFollowIndex_backlog(ShardChangesIT.java:357)
   >    at java.lang.Thread.run(Thread.java:748)
   >    Suppressed: java.lang.AssertionError: 
```

```
FAILURE 14.8s J1 | ShardChangesIT.testFollowIndex_backlog <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <6942L>
   >      but: was <7000L>
   >    at __randomizedtesting.SeedInfo.seed([7C66F5C828D24866:A6B974465F84C94E]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.xpack.ccr.ShardChangesIT.lambda$assertTotalNumberOfOptimizedIndexing$30(ShardChangesIT.java:898)
   >    at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:847)
   >    at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:821)
   >    at org.elasticsearch.xpack.ccr.ShardChangesIT.assertTotalNumberOfOptimizedIndexing(ShardChangesIT.java:882)
   >    at org.elasticsearch.xpack.ccr.ShardChangesIT.testFollowIndex_backlog(ShardChangesIT.java:353)
   >    at java.lang.Thread.run(Thread.java:748)
```

Build urls:
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+periodic/98/console

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/ShardChangesIT.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/ShardChangesIT.java
@@ -355,7 +355,7 @@ public class ShardChangesIT extends ESIntegTestCase {
 
         assertSameDocCount("index1", "index2");
         assertTotalNumberOfOptimizedIndexing(resolveIndex("index2"), numberOfShards,
-            client().prepareSearch("index2").get().getHits().totalHits);
+            client().prepareSearch("index1").get().getHits().totalHits);
         unfollowIndex("index2");
         assertMaxSeqNoOfUpdatesIsTransferred(resolveIndex("index1"), resolveIndex("index2"), numberOfShards);
     }
@@ -916,7 +916,7 @@ public class ShardChangesIT extends ESIntegTestCase {
                 for (String node : internalCluster().nodesInclude(followerIndex.getName())) {
                     IndicesService indicesService = internalCluster().getInstance(IndicesService.class, node);
                     IndexShard shard = indicesService.getShardOrNull(new ShardId(followerIndex, shardId));
-                    if (shard != null) {
+                    if (shard != null && shard.routingEntry().primary()) {
                         try {
                             FollowingEngine engine = ((FollowingEngine) IndexShardTestCase.getEngine(shard));
                             numOfOptimizedOps[shardId] = engine.getNumberOfOptimizedIndexing();

